### PR TITLE
Separate error element

### PR DESCRIPTION
**Pull request by martijnjanssen**
Completely removed all the duplicate code and recurring code. This has removed quite some code from all elements and allowing for an easier way of showing errors.

### CODE REVIEW

File: src/lancie-error/lancie-error.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        display: block;
+        width: inherit;
+        padding: 10px 20px;
+        border-left: 4px solid #F44336;
+        background: #FFCDD2;
+        color: #B71C1C;
+      }
+    </style>
+
+    [[message]]
```
Comment (TimvdLippe) [inline src/lancie-error/lancie-error.html]: Wrap this in a `<span>`

File: src/lancie-error/lancie-error.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../../bower_components/polymer/polymer.html">
+
+<dom-module id="lancie-error">
+  <template>
+    <style>
+      :host {
```
Comment (TimvdLippe) [inline src/lancie-error/lancie-error.html]: Needs extra rule to actually `display: none` on hidden. This is not propagated in shadow dom

File: src/lancie-login/lancie-login-form.html [hunk 3]
```diff
@@ -109,50 +96,37 @@
...
         this._token = request.response.object;
         this.$.getCurrentUser.generateRequest();
       },
 
-      onFailedLogin: function(event, request) {
-        this.loginError = true;
+      onLoginError: function(event, request) {
         if (request.request.status === 401) {
-          this.loginErrorMessage = 'Bad credentials.';
+          this.$.error.setError('Bad credentials.')
```
Comment (skamoen) [inline src/lancie-login/lancie-login-form.html]: While we're doing stuff here, please add an extra catch for unverified users. The request message will be "Authentication Failed: User is disabled", which should display an error like "Please validate your email first."

Comment (martijnjanssen) [inline src/lancie-login/lancie-login-form.html]: I will fix this in the PR fixing the login endpoint.

**Pull request merged**